module-2/backend/week_three.md

```diff
@@ -9,17 +9,25 @@ Note: When you're done, submit a PR with a reflection in the comments about how
 
 ### Questions
 
-1. What are the 3 main features in an ERD?
+1. What are the 3 main features in an ERD? Entities, Relationships, Cardinality
 2. Create an ERD for the following database: Restaurants, Customers, Items, Ingredients, Beverages, Orders, Vendors.
-3. What is the entry at the command line to create a new rails app?
-4. What do Models generally inherit from in rails?
-5. What do Controllers generally inherit from in a rails project?
-6. How would I create a route if I wanted to see a specific horse in my routes fitle assuming I'm sticking to standard conventions and that I didn't want other CRUD functionality?
-7. What rake task is useful when looking at routes, and what information does it give you?
-8. What is an example of a route helper? When would you use them?
-9. What's the difference between what `_url` and `_path` return when combined with a routes prefix?
-10. What are strong params and why are the necessary?
-11. What role does `form_for` play in helping us create our forms?
-12. How does `form_for` know where to submit the user's input?
+3. What is the entry at the command line to create a new rails app? Rails new
+4. What do Models generally inherit from in rails? ApplicationRecord
+5. What do Controllers generally inherit from in a rails project? ApplicationController
+6. How would I create a route if I wanted to see a specific horse in my routes fitle assuming I'm sticking to standard conventions and that I didn't want other CRUD functionality? resources :horses, only: [:show]
+7. What rake task is useful when looking at routes, and what information does it give you? rake routes. It gives you a table of all the helper paths, the actual paths, and the associated controllers and actions.
+8. What is an example of a route helper? When would you use them? in the Horses situation, horse_path(id) is a route helper that you can use to access the show page of a particular horse. You can use them within the action methods defined in your controller to control what happens under certain conditions. 
+9. What's the difference between what `_url` and `_path` return when combined with a routes prefix? _url returns the same path as _path but includes the current host, port, and path prefix. It is best practice to use _path.
+10. What are strong params and why are the necessary? Strong permit the entry of only certain types of params into a form. This prevents an attacker from maliciously using a form to corrupt the database.
+11. What role does `form_for` play in helping us create our forms? form_for is a helper method which generates a form to collect params for a particular object.
+12. How does `form_for` know where to submit the user's input? You pass in as an argument an instance variable declared in your new or edit method which points to the exact object for which the user is creating or updating.
```

Comment:
yup! and depending on whether or not this object exists in the database or not, Rails knows to send it to the `create` or `update` action